src/datadog/trace_segment.cpp

```diff
@@ -48,8 +64,11 @@ TraceSegment::TraceSegment(
     const std::vector<PropagationStyle>& injection_styles,
     const Optional<std::string>& hostname, Optional<std::string> origin,
     std::size_t tags_header_max_size,
-    std::unordered_map<std::string, std::string> trace_tags,
+    std::vector<std::pair<std::string, std::string>> trace_tags,
```

Comment:
That's a lot of parameters for a constructor. Just sayin.

Reply:
Yes, I could create a `struct` to hold them, but I decided not to. What do you think?